packages/gestalt/src/Tile/Tile.js

```diff
@@ -0,0 +1,161 @@
+// @flow strict
+import { type Node, useEffect, useState } from 'react';
+import classnames from 'classnames';
+import Checkbox from '../Checkbox.js';
+import Flex from '../Flex.js';
+import { ENTER } from '../keyCodes.js';
+import Tooltip from '../Tooltip.js';
+import useFocusVisible from '../useFocusVisible.js';
+import useInteractiveStates from '../utils/useInteractiveStates.js';
+import { FixedZIndex } from '../zIndex.js';
+import styles from './Tile.css';
+
+type Props = {|
+  /**
+   * Prop description.
+   */
+  children?: Node,
+  /**
+   * The background color of the tile as a hex. Don't forget about tokenization
+   */
+  bgColor?: string,
+  /**
+   * The border color of the tile. Don't forget about tokenization
+   */
+  borderColor?: string,
+  /**
+   * Should the tile be disabled
+   */
+  disabled?: boolean,
+  /**
+   * Id to identify the tile
+   */
+  id?: string,
+  /**
+   * Indicate if tile is in a selected state
+   */
+  selected?: boolean,
+  /**
+   * Handler if the item is selected
+   */
+  onChange?: ({| selected: boolean, id?: string |}) => void,
+  /**
+   * Ties a visible checkbox to the tile's selected state
+   */
+  showCheckbox?: boolean,
+  /**
+   * Adds a Tooltip on hover/focus of the Tile. See the With Tooltip variant to learn more.
+   */
+  tooltip?: string,
+|};
+
+function DisabledOverlay() {
+  return <div className={classnames(styles.tile, styles.disabledOverlay)} />;
+}
+
+function TooltipWrapper({
```

Comment:
nit: this name makes it sounds like this is a wrapper around Tooltip, instead of a component that conditionally wraps children with Tooltip. Consider something like MaybeTooltip or similar